### ISSUE TITLE
HWPH deleting memory

### ISSUE THREAD

**Issue by galanCA** [bigladder/HPWHsim#144]
When performing the nSubhrTicks experiment, I encounter errors on a few runs. These runs seem to be random, and the error is that hpwh is trying to access deleted data. Using visual studio debugger, we found that in [line 2949 of hpwh.cc](https://github.com/EcotopeResearch/HPWHsim/blob/master/src/HPWH.cc#L2949) hpwh is trying to access hpwhVerbosity, but all the data inside does not exist. We suspect that at some point, hpwh was deleted but did not set the pointer to NULL. Could you please look into this? 

The subHrTicks were the error occurred were the following: 9, 18, 21, 36, 39, 42, 49, 55. These correspond to a timestep of 6.666666666666667, 3.3333333333333335, 2.857142857142857, 1.6666666666666667, 1.5384615384615385, 1.4285714285714286, 1.2244897959183674, 1.0909090909090908 in minutes.

My debugging efforts were mostly focused on tick number 42, but the file can be modified to test any of the subHrTicks. Please find the CSE file attached to this issue, and it can be run using [this branch](https://github.com/cse-sim/cse/tree/user-input-subhrticks). 
[Uploading 3BR_2022-HPWH-CZ12 - Prop_42_1.zip…]()

**Comment by pmskintner** [comment]
Sounds mysterious. I have a couple clarifying questions for you. Those are some random timesteps, did you try every possible subHrTick from 1 - 60, and the ones you listed are just the ones that caused issue?

Are you deleting the HPWH object in CSE? The hpwh object in the line should be the parent for the heatsource, so I'm not deleting it. 

I'll look into it and let you know if anything else comes up.

**Comment by pmskintner** [comment]
The zip file link seems to be broken, can you upload it again?

**Comment by galanCA** [comment]
Yes, I tried every step from 1 to 60 and the ticks listed above were the ones causing this error. 

We do delete the HPWH object but only during the [destructor](https://github.com/cse-sim/cse/blob/master/src/DHWCalc.cpp#L2591-L2606). 

Attached should be the zip folder let me know if it doesn't work.

[3BR_2022-HPWH-CZ12 - Prop_42_1.zip](https://github.com/EcotopeResearch/HPWHsim/files/8732349/3BR_2022-HPWH-CZ12.-.Prop_42_1.zip)

**Comment by pmskintner** [comment]
I seem to be missing this file: 

'3BR_2022-HPWH-CZ12 - Prop-tdv.csv'

It's been little while since I've worked with CSE, so I'm not sure where I would find this file either. Can you supply it? Also just for my learning, is this file typically written out by CBECC for CSE?

**Comment by galanCA** [comment]
I added the missing files and a readme explaining how to replicate the problem. Hopefully, you are able to replicate the same results.

[Replicate nSubhrTicks Bug.zip](https://github.com/EcotopeResearch/HPWHsim/files/8758113/Replicate.nSubhrTicks.Bug.zip)

**Comment by pmskintner** [comment]
Thanks! I'm able to replicate... Now I just have to figure out why

**Comment by pmskintner** [comment]
Maybe I spoke to soon. I'm not able to replicate the issue on breaking line 2949 of hpwh.cc. 

Using visual studio 2022 debugger if I run it I get an error at the DHWSYS() destructor on [line 753 of DHWCalc.cpp](https://github.com/cse-sim/cse/blob/86b02a687668a7be393e1be2884d970a7e4702e9/src/DHWCalc.cpp#L753), when it was entered from the second call of to line [line 1509of DHWCalc.cpp](https://github.com/cse-sim/cse/blob/86b02a687668a7be393e1be2884d970a7e4702e9/src/DHWCalc.cpp#L1509).

Can you double check your error and provide anything need for me to replicate your error?

**Comment by galanCA** [comment]
I will look into this and get back to you.

**Comment by galanCA** [comment]
If you pull the latest changes for the branch you should get the original error ( [line 233 HPWH.cc](https://github.com/EcotopeResearch/HPWHsim/blob/master/src/HPWH.cc#L233) ). Let me know if you get a different error.
Thank you!

**Comment by pmskintner** [comment]
Hi I'm still getting a different error. But it has changed to the next line in topDHW(), [line 1510 of DHWCalc.cpp](https://github.com/cse-sim/cse/blob/86b02a687668a7be393e1be2884d970a7e4702e9/src/DHWCalc.cpp#L1510).

For reference here's my command line output, it looks like its getting farther than the line in the readme you sent.

CSE 0.905.0+user-input-subhrticks.fe6a5334.75 for Win32 console
Command line: 3BR_2022-HPWH-CZ12 - Prop_42_1.cse
  C:\Users\paul\Documents\GitHub\cse\issues\3BR_2022-HPWH-CZ12 - Prop_42_1.cse run 001:
    Input

C:\USERS\PAUL\DOCUMENTS\GITHUB\CSE\ISSUES\3BR_2022-HPWH-CZ12 - PROP_42_1.CSE(58): Warning:
  top: nSubhrTicks is currenlty used only for experimental purposes

    AutoSizing Pass 1 Heat 02-Jul  Pass 2 Heat 02-Jul
    Re-setup
    


Interestingly when I run the debug vs the release .exe I get an extra message: 

C:\USERS\PAUL\DOCUMENTS\GITHUB\CSE\ISSUES\3BR_2022-HPWH-CZ12 - PROP_42_1.CSE(58): Warning:
  top: nSubhrTicks is currenlty used only for experimental purposes

I'm not sure if its just repeating the warnings in debug mode or it made it farther in the processing.

**Comment by chipbarnaby** [comment]
The TDV file contains hourly Time-Dependant Values for electricity and
natural gas.  Any TDV file is sufficient for debug runs.

I've attached one, just modify tdvFName to point to it.

(Also not, you may not need a tdv file at all -- it depends on whether the
model uses TDV values.  Try just deleting tdvFName.)
